boggle.js

```diff
@@ -7,6 +7,15 @@ function newBoggleBoard() {
   ];
 }
 
+function shuffle(array) {
+  for (let x = 0; x < 16; x++) {
+    let y = Math.floor(Math.random() * (x + 1));
+    [array[x], array[y]] = [array[y], array[x]];
+  }
+}
+
+// source for shuffle: http://javascript.info/task/shuffle
```

Comment:
Your code doesn't implement this solution exactly, but I _think_ it's equivalent. I'd either implement it exactly or make sure your code is generating all possible shufflings with equal probability.

You can repurpose your code from `textalyzer` to get a frequency count of each possible reshuffling. Generate a few thousand reshufflings and see if they occur with similar likelihoods.